datastore/google/cloud/datastore/entity.py

```diff
@@ -181,6 +181,19 @@ def kind(self):
         if self.key:
             return self.key.kind
 
+    @property
+    def id(self):
+        """Get the id of the current entity.
+
+        .. note::
+          This relies entirely on the :class:`gcloud.datastore.key.Key`
+          set on the entity.  That means that we're not storing the id
+          of the entity at all, just the properties and a pointer to a
+          Key which knows its Id.
```

Comment:
I don't think this last part of the note (about storage) is necessary. It's an implementation detail.